tests/network/vmi_istio.go

```diff
@@ -0,0 +1,207 @@
+		libvmi.WithNetwork(network),
+		libvmi.WithInterface(libvmi.InterfaceDeviceWithMasqueradeBinding(ports...)),
+		withIstioAnnotation(),
+		withAppLabel(appLabel),
+	)
+	tests.AddServiceAccountDisk(vmi, "default")
+	return vmi
+}
+
+var _ = SIGDescribe("[Serial] Istio", func() {
+	var (
+		err        error
+		virtClient kubecli.KubevirtClient
+	)
+	network := &v1.Network{
+		Name: "default",
+		NetworkSource: v1.NetworkSource{
+			Pod: &v1.PodNetwork{},
+		},
+	}
+	explicitPorts := []v1.Port{
+		{Port: svcDeclaredTestPort},
+		{Port: svcUndeclaredTestPort},
+	}
+	vmiAppCases := map[string]*vmiAppCase{
+		vmiDefaultPortsAppCaseName: {
+			appSelector: vmiDefaultPortsAppSelector,
+			ports:       []v1.Port{},
+		},
+		vmiExplicitPortsAppCaseName: {
+			appSelector: vmiExplicitPortsAppSelector,
+			ports:       explicitPorts,
+		},
+	}
+
+	Describe("Virtual machine in a service mesh using masquerade interface with explicit port list", func() {
+		tests.BeforeAll(func() {
+			if !istioServiceMeshDeployed() {
+				Skip("Istio service mesh is required for service-mesh tests to run")
+			}
+
+			tests.BeforeTestCleanup()
+			virtClient, err = kubecli.GetKubevirtClient()
+			tests.PanicOnError(err)
+
+			tests.AddLabelToNamespace(tests.NamespaceTestDefault, istioInjectNamespaceLabel, "enabled")
+
+			for appCaseKey, vmiCase := range vmiAppCases {
+				vmi := newRandomCirrosVMIWithIstioSidecarAndDefaultServiceAccountAndNetworkAndInterface(
+					vmiCase.appSelector,
+					network,
+					vmiCase.ports,
+				)
+				svc := newServiceWithAppSelector(
+					fmt.Sprintf("%s-service", vmi.Name),
+					vmi.Namespace,
+					vmiCase.appSelector,
+				)
+				vmiAppCases[appCaseKey].vmi = vmi
+				vmiAppCases[appCaseKey].service = svc
+
+				_, err = virtClient.CoreV1().Services(svc.Namespace).Create(context.Background(), svc, metav1.CreateOptions{})
+				Expect(err).ShouldNot(HaveOccurred())
+				_, err = virtClient.VirtualMachineInstance(vmi.Namespace).Create(vmi)
+				Expect(err).ToNot(HaveOccurred())
+			}
+			for _, vmiCase := range vmiAppCases {
+				tests.WaitUntilVMIReady(vmiCase.vmi, libnet.WithIPv6(console.LoginToCirros))
+				for _, targetPort := range []int{svcDeclaredTestPort, svcUndeclaredTestPort} {
+					tests.StartHTTPServer(vmiCase.vmi, targetPort)
+				}
+			}
+		})
+		table.DescribeTable("should be able to reach", func(destination string, port int) {
+			var srcVMI, dstVMI *v1.VirtualMachineInstance
+			switch destination {
+			case vmiExplicitPortsAppCaseName:
+				srcVMI = vmiAppCases[vmiDefaultPortsAppCaseName].vmi
+				dstVMI = vmiAppCases[vmiExplicitPortsAppCaseName].vmi
+			case vmiDefaultPortsAppCaseName:
+				srcVMI = vmiAppCases[vmiExplicitPortsAppCaseName].vmi
+				dstVMI = vmiAppCases[vmiDefaultPortsAppCaseName].vmi
+			}
+			dstVMI, err := virtClient.VirtualMachineInstance(dstVMI.Namespace).Get(dstVMI.Name, &metav1.GetOptions{})
+			Expect(err).ShouldNot(HaveOccurred())
+			dstVMIAddress := dstVMI.Status.Interfaces[0].IP
```

Comment:
Missing ipv6 tests.
[Example](https://github.com/kubevirt/kubevirt/blob/876f1bec44659b8f3ee0784d6ac4a48c37f48a08/tests/vmi_networking_test.go#L668) of how to test ipv6.

Reply:
IPv6 is not supported by Istio, so testing ipv6 is not relevant IMO

Reply:
[done](https://github.com/kubevirt/kubevirt/compare/e47924b601c29ede8f8b175d6ede514fe9f480bb..2a20abaa3692d559d53ec3a034cd7465209461f0)